src/Chainweb/CutDB.hs

```diff
@@ -146,6 +148,7 @@ defaultCutDbConfig v = CutDbConfig
 data CutDb cas = CutDb
     { _cutDbCut :: !(TVar Cut)
     , _cutDbQueue :: !(PQueue (Down CutHashes))
+    , _cutNetworkCutHeight :: !(TVar (Maybe BlockHeight))
```

Comment:
Should the name of this field somehow indicate that this is just an *expected* value, or an educated guess of what the consensus is. For instance `approxNetworkCutHeight` or `expectedMinNetworkCutHeight`?

Reply:
While using this information can greatly improve performance, it has also great potential for being abused. It gives an attacker the ability to remotely switch off mining, by sending invalid cuts of large block height.I think, the name should somehow reflect that it should be used with care.

Going forward, we may factor in the expected block height based on block rate and time of genesis to predict if the expectation is reasonable. And, of course, we should add some sort of reputation management, but that alone wouldn't be enough. Once we have persistence implemented, we'll probably want a configuration option to turn this feature on and of manually.

Reply:
What about using the average (or some regression) of all cuts in the queue? That could mitigate some of the risk. Well, we would still have to make sure that the cuts in the queue represent a good mix of origin, for which we would need some Kademlia-style partitioning/classification of nodes (both for GET and PUT).

Reply:
I've done the renaming.